### DISCUSSION TITLE
Preserve function name

### DISCUSSION THREAD

**Discussion by ramosjeremy** [javascript-obfuscator/javascript-obfuscator#1145] [Q&A]
I have a function named eventCapture(). I would like to keep the name of the function unchanged so that I can reference it in another segment of code. Is there a way to do that, or is there a way to know that the new symbol is for the function's name?

**Comment by BlackYuzia** [comment]
You could use whitelist to save your function(s) name. If I right understand what do you want.

**Comment by ramosjeremy** [comment]
Suppose I have a file with 

function eventCapture(){
// Some logic that I want obfuscated here
return result
}

Then I have another file that will be calling eventCatpure. This second file cannot be obfuscated.

div.onclick = eventCapture

How can I obfuscate the body of the eventCapture function, so that I can reference it as above?

**Reply by relative** [reply]
add [`reservedNames: ['eventCapture']`](https://github.com/javascript-obfuscator/javascript-obfuscator#reservednames) to your obfuscator configuration

**Reply by ramosjeremy** [reply]
That worked. Thank you so much.
